packages/next/next-server/lib/router/router.ts

```diff
@@ -17,10 +17,15 @@ import { isDynamicRoute } from './utils/is-dynamic'
 import { getRouteMatcher } from './utils/route-matcher'
 import { getRouteRegex } from './utils/route-regex'
 
-function addBasePath(path: string): string {
-  // @ts-ignore variable is always a string
-  const p: string = process.env.__NEXT_ROUTER_BASEPATH
-  return path.indexOf(p) !== 0 ? p + path : path
+// @ts-ignore variable is always a string
+const basePath: string = process.env.__NEXT_ROUTER_BASEPATH
```

Comment:
Maybe instead of `// @ts-ignore`?
```suggestion
const basePath: string = process.env.__NEXT_ROUTER_BASEPATH as string
```

Reply:
I'm gonna leave the `@ts-ignore` since we probably want comment here anyways

Reply:
it will silence every possible typescript warning on the following line, not only the one you don't care about. in general you'd want to avoid ts-ignore whenever you can. you can still keep the comment.